engine/crates/engine-v2/src/plan/logical/graph.rs

```diff
@@ -0,0 +1,93 @@
+use super::LogicalPlan;
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Hash)]
+pub struct PlanId(usize); // should be a u16, more than enough for any sane query
```

Comment:
I'd vote u32 but sure

Reply:
removed the comment, we'll see it another day. Given how the plans are currently generated I think 65K is more than enough. But it's a decision that can be easily changed later. 🤷